Core/src/Core/Filters/ConvertDataFilter.cpp

```diff
@@ -0,0 +1,168 @@
+std::vector<std::string> ConvertDataFilter::defaultTags() const
+{
+  return {"#Core", "#Convert"};
+}
+
+//------------------------------------------------------------------------------
+Parameters ConvertDataFilter::parameters() const
+{
+  Parameters params;
+
+  params.insertSeparator(Parameters::Separator{"Input Parameters"});
+  params.insert(std::make_unique<NumericTypeParameter>(k_ScalarType_Key, "Scalar Type", "", NumericTypeParameter::ValueType::int8));
+  params.insert(std::make_unique<ArraySelectionParameter>(k_ArrayToConvert_Key, "Data Array to Convert", "", DataPath{}, GetAllDataTypes()));
+  params.insertSeparator(Parameters::Separator{"Output Parameters"});
+  params.insert(std::make_unique<DataObjectNameParameter>(k_ConvertedArray_Key, "Converted Data Array", "", "Converted_"));
+
+  return params;
+}
+
+//------------------------------------------------------------------------------
+IFilter::UniquePointer ConvertDataFilter::clone() const
+{
+  return std::make_unique<ConvertDataFilter>();
+}
+
+//------------------------------------------------------------------------------
+IFilter::PreflightResult ConvertDataFilter::preflightImpl(const DataStructure& dataStructure, const Arguments& filterArgs, const MessageHandler& messageHandler,
+                                                          const std::atomic_bool& shouldCancel) const
+{
+  auto pScalarType = filterArgs.value<NumericTypeParameter::ValueType>(k_ScalarType_Key);
+  auto pInputArrayPath = filterArgs.value<ArraySelectionParameter::ValueType>(k_ArrayToConvert_Key);
+  auto pConvertedArrayName = filterArgs.value<std::string>(k_ConvertedArray_Key);
+  DataPath convertedArrayPath = pInputArrayPath.getParent().createChildPath(pConvertedArrayName);
+
+  // Declare the preflightResult variable that will be populated with the results
+  // of the preflight. The PreflightResult type contains the output Actions and
+  // any preflight updated values that you want to be displayed to the user, typically
+  // through a user interface (UI).
+  PreflightResult preflightResult;
+
+  // If your filter is making structural changes to the DataStructure then the filter
+  // is going to create OutputActions subclasses that need to be returned. This will
+  // store those actions.
+  complex::Result<OutputActions> resultOutputActions;
+
+  // If your filter is going to pass back some `preflight updated values` then this is where you
+  // would create the code to store those values in the appropriate object. Note that we
+  // in line creating the pair (NOT a std::pair<>) of Key:Value that will get stored in
+  // the std::vector<PreflightValue> object.
+  std::vector<PreflightValue> preflightUpdatedValues;
+
+  auto* inputArray = dataStructure.getDataAs<IDataArray>(pInputArrayPath);
+  if(inputArray == nullptr)
+  {
+    return {nonstd::make_unexpected(std::vector<Error>{Error{-15201, fmt::format("Cannot find input data array at path '{}'", pInputArrayPath.toString())}})};
+  }
+
```

Comment:
I feel like switch statement could be reduced to just setting a local variable (targetNumericType) to the proper NumericType::XXXX value.  

Then just have a single line where the ```resultOutputActions.value().actions.push_back(
        std::make_unique<CreateArrayAction>(targetNumericType, inputArray->getIDataStoreRef().getTupleShape(), inputArray->getIDataStoreRef().getComponentShape(), convertedArrayPath));``` is called

Reply:
@imikejackson The CreateArrayAction actually takes a DataType enum value though and not a NumericType.  They are 2 different enums.  The parameter makes use of the NumericType enum but the action makes use of the DataType enum.  This sort of brings out another point that I wasn't sure of how to handle; so the NumericType enum does not have a bool option which was supported in SIMPL, so perhaps a NumericTypeParameter isn't the best option here?